src/main.rs

```diff
@@ -258,7 +263,7 @@ impl PreparedStatements {
         block_time: Timestamp,
         block_height: AbsoluteBlockHeight,
         ts: &BlockItemSummaryWithCanonicalAddresses,
-    ) -> Result<(), postgres::Error> {
+    ) -> Result<(), DatabaseError> {
```

Comment:
As our `Unkown` error is not really a `DatabaseError` (the error doesn't come from the database), I would create a new error type e.g.:

```
/// A collection of possible errors that can happen while indexing blocks into the database.
#[derive(Debug, Error)]
pub enum IndexingError {
    /// Database error.
    #[error("Error using the database {0}.")]
    PostgresError(#[from] postgres::Error),
    /// ...
    #[error("Please update the rust SDK. Type {0} is unkown.")]
    Unkown(String),
}
```

It is a weird that this very general error case is in the type.
```  
  /// Other errors while processing database data.
    #[error("Error happened on database thread {0}.")]
    OtherError(#[from] anyhow::Error),
   ```